src/jit/lsra.cpp

```diff
@@ -8260,10 +8260,49 @@ void LinearScan::resolveEdge(BasicBlock*      fromBlock,
             location[sourceReg]        = REG_NA;
 
             // Do we have a free targetReg?
-            if (fromReg == sourceReg && source[fromReg] != REG_NA)
+            if (fromReg == sourceReg)
             {
-                regMaskTP fromRegMask = genRegMask(fromReg);
-                targetRegsReady |= fromRegMask;
+                if (source[fromReg] != REG_NA)
+                {
+                    regMaskTP fromRegMask = genRegMask(fromReg);
+                    targetRegsReady |= fromRegMask;
+#ifdef _TARGET_ARM_
+                    if (genIsValidDoubleReg(fromReg))
+                    {
+                        // Ensure that the other half is free.
+                        Interval* otherInterval = sourceIntervals[source[fromReg]];
+                        regNumber upperHalfReg  = REG_NEXT(fromReg);
+                        if ((otherInterval->registerType == TYP_DOUBLE) && (location[upperHalfReg] != REG_NA))
+                        {
+                            targetRegsReady &= ~fromRegMask;
```

Comment:
> When an even float register becomes free, we can't add it to targetRegsReady unless it's other half is also free.

For example we have such resolution:
`f2->f1, f3->f2.` (in the end of the block01 LV01 lives on f2, LV02 lives on f3, in the beginning of the block02 LV01 lives on f1, LV02 lives on f2 and we have an edge from BB01 to BB02).
we would move `f3` to `f2`, but do not add `f2` to `targetRegsReady`, am I right?
It means we will have to spill something then?

Reply:
We would first move f2 to f1, since f1 is free. Then f2 would be free so we would move f3 to f2. We couldn't move f3 to f2 first because f2 is occupied. So maybe I don't understand your example. The only time we have to spill or swap is if we have a circular dependency.

Reply:
>We couldn't move f3 to f2 first because f2 is occupied.

Yes, we should move f2 to f1 first.

>Then f2 would be free so we would move f3 to f2. 

Will be f2 free with the new code? I think we will check that its other half (f3) is not free and we will do ` targetRegsReady &= ~fromRegMask;` that deletes f2 from free set.

Reply:
The lclVar in f3 cannot be a double (otherwise it would be in an even register). So we won't hit this case, because `otherInterval->registerType` is not `TYP_DOUBLE`.

Reply:
Got it, thank you. Could you please change `// Ensure that the other half is free. ` to something like `// Ensure that it is not a part of a double or the other half of the double is free. `

Reply:
Done.